sunpy/spectra/spectrum.py

```diff
@@ -37,6 +37,8 @@ def __new__(cls, data, *args, **kwargs):
 
     def __init__(self, data, freq_axis):
         self.freq_axis = freq_axis
+        if np.shape(data)[0] != np.shape(freq_axis)[0]:
```

Comment:
Yes, I ran both spectrum.py and spectrogram.py, it runs, without crashing.
Do I finally push it into this PR then?